### DISCUSSION TITLE
Error querying the database: db error: ERROR: syntax error at or near "ALTER"

### DISCUSSION THREAD

**Discussion by Tobbe** [prisma/prisma#6637] [Q&A] [answered]
When I try to apply one of my migrations (in prod) I get this error

```
12:46:54 AM: Error: Database error
12:46:54 AM: Error querying the database: db error: ERROR: syntax error at or near "ALTER"
12:46:54 AM:    0: migration_core::api::ApplyMigrations
12:46:54 AM:              at migration-engine/core/src/api.rs:73
12:46:54 AM: error Command failed with exit code 1.
```

The "ALTER" line in my migration looks like this
```sql
ALTER TABLE "Team" ADD COLUMN     "club" TEXT NOT NULL DEFAULT E'';
```

If I manually run that line against my database it works just fine.

How can I make my migration go through?

**Comment by Tobbe** [comment, marked as answer]
Had a missing `;` in the migration file. After adding it, everything works again.

**Reply by janpio** [reply]
Interesting - in the statement before?
We might want to think about helping our users better to validate their SQL.

**Reply by Tobbe** [reply]
Yeah, exactly. Statement before.

I'm on Windows, and I'm using DBeaver to manage my DBs. Pasting the migration file (with missing semicolon) and running it works just fine.
My colleague is on Mac and is using Table Plus. He could also run the SQL from the migration manually without problem.

So, yeah, you could validate the SQL and throw helpful error messages. Or you could be more lenient and allow missing semicolons as our two DB gui clients seem to do :)

**Reply by janpio** [reply]
I think the DB GUIs are probably splitting the files by each statement and then executing them individually - then semicolons do not matter.

cc @tomhoule

**Reply by Tobbe** [reply]
Yes, that's what it looks like in DBeaver at least

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
